cartographer/cloud/map_builder_server_main.cc

```diff
@@ -41,6 +41,7 @@ void Run(const std::string& configuration_directory,
 #if USE_PROMETHEUS
   metrics::prometheus::FamilyFactory registry;
   ::cartographer::metrics::RegisterAllMetrics(&registry);
+  MapBuilderServerInterface::RegisterMetrics(&registry);
```

Comment:
Calling a static method of an interface looks a bit confusing to me. Maybe, move `#if USE_PROMETHES` after
```
 std::unique_ptr<MapBuilderServerInterface> map_builder_server =
      CreateMapBuilderServer(map_builder_server_options,
                             std::move(map_builder));
```
and use `map_builder_server.RegisterMetrics` instead. 

Or maybe move  `MapBuilderServerInterface::RegisterMetrics` outside of the interface, but keep it in the same header. It can be renamed to 
```
namespace metrics { 
  RegisterMapBuilderServer(metrics::FamilyFactory* factory){}
}
```

Reply:
Haha, `RegisterMapBuilderServerMetrics` was what I originally had but I did not like the long function name so I moved it into the interface. I guess that's a matter of taste. Ok let's go with that.